### PR TITLE
Completed task requirements

### PR DESCRIPTION
**Pull request by Likh-Alex**

### CODE REVIEW

File: src/main/java/core/mate/academy/model/BulldozerProducer.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer<Bulldozer> {
+    @Override
+    public List<Bulldozer> get() {
+        List<Bulldozer> bulldozers = new ArrayList<>();
+        bulldozers.add(new Bulldozer("Mull", "Black", 200));
+        bulldozers.add(new Bulldozer("Donk", "Pink", 300));
+        bulldozers.add(new Bulldozer("Kong", "Yellow", 400));
+        return bulldozers;
```
Comment (gigbat) [inline src/main/java/core/mate/academy/model/BulldozerProducer.java:14]: You can return immediately using List.of(new Bulldozer("Mull", "Black", 200), ... )

File: src/main/java/core/mate/academy/model/ExcavatorProducer.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Excavator> {
+    @Override
+    public List<Excavator> get() {
+        List<Excavator> excavators = new ArrayList<>();
+        excavators.add(new Excavator("Handler", "Brown", 5));
+        excavators.add(new Excavator("Grabber", "Red", 4));
+        excavators.add(new Excavator("Bandit", "White", 6));
+        return excavators;
```
Comment (gigbat) [inline src/main/java/core/mate/academy/model/ExcavatorProducer.java:14]: read the previous comment

File: src/main/java/core/mate/academy/model/TruckProducer.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Truck> {
+    @Override
+    public List<Truck> get() {
+        List<Truck> trucks = new ArrayList<>();
+        trucks.add(new Truck("Mighty", "Green", 200));
+        trucks.add(new Truck("Qwerty", "Orange", 400));
+        trucks.add(new Truck("Enforcer", "Blue", 500));
+        return trucks;
```
Comment (gigbat) [inline src/main/java/core/mate/academy/model/TruckProducer.java:14]: .

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 4]
```diff
@@ -1,7 +1,43 @@
...
+            return new ArrayList<>(excavators);
+        }
+        List<? extends Machine> trucks = new TruckProducer().get();
+        return new ArrayList<>(trucks);
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        int size = machines.size();
+        for (int i = 0; i < size; i++) {
```
Comment (gigbat) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:32]: Don't create a new variable. Immediately pass the machines.size() to the loop

**Review by gigbat — CHANGES_REQUESTED** [review]

File: src/main/java/core/mate/academy/model/BulldozerProducer.java (hunk 1 above)
Comment (Likh-Alex) [reply src/main/java/core/mate/academy/model/BulldozerProducer.java]: Thank you, done

File: src/main/java/core/mate/academy/model/ExcavatorProducer.java (hunk 2 above)
Comment (Likh-Alex) [reply src/main/java/core/mate/academy/model/ExcavatorProducer.java]: done here as well

File: src/main/java/core/mate/academy/model/TruckProducer.java (hunk 3 above)
Comment (Likh-Alex) [reply src/main/java/core/mate/academy/model/TruckProducer.java]: and here also

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java (hunk 4 above)
Comment (Likh-Alex) [reply src/main/java/core/mate/academy/service/MachineServiceImpl.java]: done

**Review by gigbat — APPROVED** [review]

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 5]
```diff
@@ -1,7 +1,42 @@
...
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            List<? extends Machine> bulldozers = new BulldozerProducer().get();
```
Comment (Sofasmile) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:19]:
```suggestion
            List<Bulldozer> bulldozers = new BulldozerProducer().get();
```
the same with other lists

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 6]
```diff
@@ -1,7 +1,42 @@
...
+            return new ArrayList<>(bulldozers);
+        } else if (type == Excavator.class) {
```
Comment (Sofasmile) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:21]: do we need `else` block after `return` statement?

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 7]
```diff
@@ -1,7 +1,42 @@
...
+            List<? extends Machine> excavators = new ExcavatorProducer().get();
+            return new ArrayList<>(excavators);
+        }
+        List<? extends Machine> trucks = new TruckProducer().get();
+        return new ArrayList<>(trucks);
```
Comment (Sofasmile) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:26]: let's add this code to the `if` block
and if there is no class type - return an empty list

**Review by Sofasmile — CHANGES_REQUESTED** [review]
good job

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java (hunk 5 above)
Comment (Likh-Alex) [reply src/main/java/core/mate/academy/service/MachineServiceImpl.java]: done

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java (hunk 6 above)
Comment (Likh-Alex) [reply src/main/java/core/mate/academy/service/MachineServiceImpl.java]: removed `else` blocks for other machines and put every machine in a separate `if` block

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java (hunk 7 above)
Comment (Likh-Alex) [reply src/main/java/core/mate/academy/service/MachineServiceImpl.java]: done

**Review by Sofasmile — APPROVED** [review]
great)

**Pull request closed without merging**